### PR TITLE
[SHELL32][SDK] Implement Activate_RunDLL

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Implementing missing features...
JIRA issue: N/A

## Proposed changes

- Add `Activate_RunDLL` function implementation.
- Add `Activate_RunDLL` prototype to `<undocshell.h>`.

## TODO

- [x] Do build.

### CODE REVIEW

File: dll/win32/shell32/utils.cpp [hunk 1]
```diff
@@ -504,3 +504,29 @@ SHOpenPropSheetA(
...
+ *
+ * Every parameter is unused.
+ */
+EXTERN_C
+BOOL WINAPI
+Activate_RunDLL(
+    DWORD dwUnused1,
+    LPVOID lpUnused2,
+    LPVOID lpUnused3,
+    LPVOID lpUnused4)
```
Comment (HBelusca) [inline dll/win32/shell32/utils.cpp:519]:
```suggestion
    _In_ DWORD dwUnused1,
    _In_ LPVOID lpUnused2,
    _In_ LPVOID lpUnused3,
    _In_ LPVOID lpUnused4)
```

Comment (HBelusca) [reply dll/win32/shell32/utils.cpp:519]: OK yeah I don't know whether this deserves an annotation at all, seeing what this is...

File: dll/win32/shell32/utils.cpp [hunk 2]
```diff
@@ -504,3 +504,29 @@ SHOpenPropSheetA(
     return SHOpenPropSheetW(pszCaptionW, ahKeys, cKeys, pclsidDefault,
                             pDataObject, pShellBrowser, pszStartPageW);
 }
+
+/*************************************************************************
+ *  Activate_RunDLL [SHELL32.105]
+ *
+ * Every parameter is unused.
```
Comment (HBelusca) [inline dll/win32/shell32/utils.cpp:511]: Can you at least tell what this function seems to be used for?

Comment (katahiromz) [reply dll/win32/shell32/utils.cpp]: Fixed.

```c
/*************************************************************************
 *  Activate_RunDLL [SHELL32.105]
 *
 * Unlocks the foreground window and allows the shell window to become the
 * foreground window. Every parameter is unused.
 */
```

**Comment by whindsaks** [comment]
This feels like an internal shell feature so where is the other part of the shell that actually needs to call this?

**Comment by whindsaks** [comment]
The name gives this away as a rundll32 function so your prototype is wrong, the first parameter is pointer sized and the last is is `int`.

**Comment by katahiromz** [comment]
> This feels like an internal shell feature so where is the other part of the shell that actually needs to call this?

This function resets the lock of foreground window by `LockSetForegroundWindow` function. Actually, some malwares are using this feature:

https://vms.drweb.co.jp/virus/?i=16663715

> The name gives this away as a rundll32 function so your prototype is wrong, the first parameter is pointer sized and the last is is int.

I will fix it.

**Comment by katahiromz** [comment]
Fixed: https://github.com/reactos/reactos/commit/a52c713755dca32bd4da647a4a56c416e5ed7ad4

**Pull request merged**